ProcessLib/SourceTerms/CreateSourceTerm.cpp

```diff
@@ -22,35 +22,60 @@ namespace ProcessLib
 std::unique_ptr<SourceTerm> createSourceTerm(
     const SourceTermConfig& config,
     const NumLib::LocalToGlobalIndexMap& dof_table_bulk,
-    const MeshLib::Mesh& mesh, const int variable_id,
+    const MeshLib::Mesh& source_term_mesh, const int variable_id,
     const unsigned integration_order, const unsigned shapefunction_order,
     std::vector<std::unique_ptr<ProcessLib::ParameterBase>> const& parameters)
 {
     //! \ogs_file_param{prj__process_variables__process_variable__source_terms__source_term__type}
     auto const type = config.config.peekConfigParameter<std::string>("type");
 
+
+    MeshLib::MeshSubset source_term_mesh_subset(source_term_mesh,
+                                                source_term_nodes);
+
+
     if (type == "Nodal")
     {
+        // Create local DOF table from the source term mesh subset for the
+        // given variable and component id.
+        std::unique_ptr<NumLib::LocalToGlobalIndexMap> dof_table_source_term(
+            dof_table_bulk.deriveBoundaryConstrainedMap(
+                variable_id, {*config.component_id},
+                std::move(source_term_mesh_subset)));
         return ProcessLib::createNodalSourceTerm(
-            config.config, config.mesh, dof_table_bulk, mesh.getID(),
-            variable_id, *config.component_id, parameters);
+            config.config, config.mesh, std::move(dof_table_source_term),
+            source_term_mesh.getID(), variable_id, *config.component_id,
+            parameters);
     }
 
     if (type == "Volumetric")
     {
+        // Create local DOF table from the source term mesh subset for the
+        // given variable and component id.
+        std::unique_ptr<NumLib::LocalToGlobalIndexMap> dof_table_source_term(
+            dof_table_bulk.deriveBoundaryConstrainedMap(
+                variable_id, {*config.component_id},
+                std::move(source_term_mesh_subset)));
         return ProcessLib::createVolumetricSourceTerm(
-            config.config, config.mesh, dof_table_bulk, parameters,
-            integration_order, shapefunction_order, variable_id,
+            config.config, config.mesh, std::move(dof_table_source_term),
+            parameters, integration_order, shapefunction_order, variable_id,
             *config.component_id);
     }
 
     if (type == "Python")
     {
 #ifdef OGS_USE_PYTHON
+        // Create local DOF table from the source term mesh subset for the
+        // given variable and component id.
```

Comment:
In Python probably for all vars/comps.
Probably, these comments can all be removed. They just describe the already very descriptive line of code that follows them.